### PR TITLE
feat: add health potion

### PR DESCRIPTION
**Pull request by JoenvandeVorle**
## Description

Adds a collectible health potion, that increases player health when used. 
Also made a few changes to the `Inventory` to fix a bug with picking up multiple of the same item.

## Testability

Open the `HealthPotion` scene to pick up the potions and use them. There is no UI element for health yet so you will need to use Debug statements to see your health.
There is also a `HealthPotionTestScene` for an aplib test.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Scripts/Items/HealthPotion.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            _playerHealth = GameObject.FindGameObjectWithTag("Player").GetComponent<HealthComponent>();
+        }
+
+        /// <summary>
+        /// Uses the health potion to heal the player.
+        /// </summary>
+        public override void UseItem()
+        {
+            base.UseItem();
+            _playerHealth.ReduceHealth(-_healAmount);
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Scripts/Items/HealthPotion.cs:28]: Dit vind ik een beetje gekke manier om een player extra health te geven. Zou je misschien een method `IncreaseHealth` toe kunnen voegen? Al is het maar een wrapper die dit doet, voor anderen zal dit beter aanvoelen denk.

**Review by Tboefijn — CHANGES_REQUESTED** [review]
Looks good. Left a comment and could you quickly try the solution discussed for PickupableComponent?

**Review by Tboefijn — DISMISSED** [review]
LGTM!

File: aplib.net-demo/Assets/Scripts/Inventory/Inventory.cs [hunk 2]
```diff
@@ -43,15 +52,23 @@ public void PickUpItem(Item item)
                 if (!item.stackable)
                     return;
 
-                _tempItemList[i].uses += item.startUses;
+                _tempItemList[i].uses++;
```
[the remark this answers is not in the document]
Comment (JoenvandeVorle) [reply aplib.net-demo/Assets/Scripts/Inventory/Inventory.cs:55]: Well, it's called startUses, so it sounds like the amount of uses you get on the first pickup (that's what I added in the summary for startUses anyway). If it's supposed to be something different, I would give it a different name like `usesAddedPerPickup`. Whatever you think is best

File: aplib.net-demo/Assets/Scripts/Items/PickupableItem.cs [hunk 3]
```diff
@@ -25,8 +25,8 @@ private void OnTriggerEnter(Collider other)
 
             Item item = gameObject.GetComponent<Item>();
 
-            inventory.PickUpItem(item);
             Destroy(gameObject);
+            inventory.PickUpItem(item);
```
[the remark this answers is not in the document]
Comment (JoenvandeVorle) [reply aplib.net-demo/Assets/Scripts/Items/PickupableItem.cs:29]: Not an issue introduced in this PR, issue moved to separate ticket.

**Review by Tboefijn — DISMISSED** [review]
LGTM!

File: aplib.net-demo/Assets/Scripts/Inventory/Item.cs [hunk 4]
```diff
@@ -7,21 +7,48 @@
...
+    /// <summary>
+    /// The amount of uses the item currently has.
+    /// </summary>
     public float uses;
-    public float startUses;
+
+    /// <summary>
+    /// The amount of uses that get added to the item when it is picked up.
+    /// </summary>
+    public float usesAddedPerPickup;
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Inventory/Item.cs:25]: I think this should be an integer o.o

File: aplib.net-demo/Assets/Scripts/Inventory/Item.cs [hunk 5]
```diff
@@ -7,21 +7,48 @@
...
+    /// Since the inventory works by comparing the name of the item,
+    /// all instances of the same item should have the same name.
+    /// </summary>
+    [SerializeField]
+    protected string _itemName;
+
+    /// <summary>
+    /// The amount of uses the item currently has.
+    /// </summary>
     public float uses;
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Inventory/Item.cs:20]: should be an integer

File: aplib.net-demo/Assets/Scripts/Items/HealthPotion.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        /// <summary>
+        /// Uses the health potion to heal the player.
+        /// </summary>
+        public override void UseItem()
+        {
+            base.UseItem();
+            _playerHealth.IncreaseHealth(_healAmount);
+            Debug.Log("current uses: " + uses);
```
Comment (QuakeEye) [inline aplib.net-demo/Assets/Scripts/Items/HealthPotion.cs:29]: OEI

**Review by Tboefijn — APPROVED** [review]
LGTM!

**Pull request merged**